pkg/boost.go

```diff
@@ -90,29 +97,42 @@ func (as *DefaultBoost) SubmitBlock(ctx context.Context, msg *capella.SubmitBloc
 		maxTipTxn := _txn
 		for _, btxn := range msg.ExecutionPayload.Transactions {
 			var txn types.Transaction
-			txn.UnmarshalBinary(btxn)
+			err := txn.UnmarshalBinary(btxn)
+			if err != nil {
+				as.config.Log.WithError(err).Error("Failed to decode transaction")
+			}
 			// Extract Min/Max
 			if txn.GasTipCapCmp(&minTipTxn) < 0 {
 				minTipTxn = txn
 			}
 			if txn.GasTipCapCmp(&maxTipTxn) > 0 {
 				maxTipTxn = txn
 			}
+
+			from, err := types.Sender(types.LatestSignerForChainID(txn.ChainId()), &txn)
+			if err != nil {
+				as.config.Log.WithField("transaction", txn).WithError(err).Error("umnable to decode sender of transaction")
+			}
+			clientID := from.Hex()
+			if blockMetadata.SearcherTxns[clientID] == nil {
```

Comment:
It's better to use this to check if key exists:
`if _, ok := blockMetadata.SearcherTxns[clientID]; !ok {`